packages/libraries/cli/src/base-command.ts

```diff
@@ -45,26 +68,66 @@ export default abstract class BaseCommand<T extends typeof Command> extends Comm
     const { args, flags } = await this.parse({
       flags: this.ctor.flags,
       baseFlags: (super.ctor as typeof BaseCommand).baseFlags,
+      enableJsonFlag: this.ctor.enableJsonFlag,
       args: this.ctor.args,
       strict: this.ctor.strict,
     });
-    this.flags = flags as Flags<T>;
-    this.args = args as Args<T>;
+    this.flags = flags as Flags<$Command>;
+    this.args = args as Args<$Command>;
   }
 
-  success(...args: any[]) {
+  /**
+   * Helper function for creating data that adheres to the type specified by your command's {@link BaseCommand.SuccessSchema}.
+   *
+   * If {@link BaseCommand.SuccessSchemaValidationEnabled} is `true`, then the given data will be runtime-validated too.
+   *
+   * For ease of use some standard properties are added for you automatically, simplifying the input you have to provide.
+   */
+  successData(dataInput: InferSuccessDataInput<$Command>): InferSuccessDataOutput<$Command> {
+    const dataOutput = {
+      ...(dataInput as object),
+      ok: true,
+      // warnings: [] as string[],
+    } as InferSuccessDataOutput<$Command>;
+
+    if (this.SuccessSchemaValidationEnabled) {
+      // TS doesn't support static property access on this.constructor for some reason.
+      const schema = (this.constructor as typeof BaseCommand).SuccessSchema as OutputType;
+      const result = schema.safeParse(dataOutput);
+      if (!result.success) {
+        throw new Errors.CLIError(result.error.message);
+      }
+      return result.data as InferSuccessDataOutput<$Command>;
+    }
+
+    return dataOutput;
+  }
+
+  /**
+   * {@link Command.log} with success styling.
+   */
+  logSuccess(...args: any[]) {
     this.log(colors.green('✔'), ...args);
   }
 
-  fail(...args: any[]) {
+  /**
+   * {@link Command.log} with failure styling.
+   */
+  logFail(...args: any[]) {
     this.log(colors.red('✖'), ...args);
   }
 
-  info(...args: any[]) {
+  /**
+   * {@link Command.log} with info styling.
+   */
+  logInfo(...args: any[]) {
     this.log(colors.yellow('ℹ'), ...args);
   }
 
-  infoWarning(...args: any[]) {
+  /**
+   * {@link Command.log} with warning styling.
+   */
+  logWarning(...args: any[]) {
```

Comment:
I renamed these to make our commands more readable. For example `this.fail`  versus `this.error` have very different behaviour but that's not clear, nor is it easy to remember which is which.